libs/ledger/src/chain/block_coordinator.cpp

```diff
@@ -912,6 +916,13 @@ BlockCoordinator::State BlockCoordinator::OnPostExecBlockValidation()
       dag_->CommitEpoch(current_block_->body.dag_epoch);
     }
 
+    // Notify notarisation of new valid block
+    // TODO(JMW): Check block is within notarisation window and previous block is notarised?
+    if (notarisation_)
+    {
+      notarisation_->NotariseBlock(current_block_->body);
+    }
+
```

Comment:
Review comment from previous PR "Looks like this should be next_block_ here, not current_block_ - as an aside, I didn't realise that even with upgraded consensus, it would still do PoW - I would suggest to rewrite it as:" - I don't think it should be next_block here. We want to notarise the block that has just been validated and executed. Also, did not add in suggested lines as I did not understand how they fit in here.

Reply:
Sorry, I see. That second one is actually a mistake as I misunderstood the point of the two pipes and have removed the notarisation from that part.

Reply:
I think you would want to notarize there though, since this is your HEAD.

Probably best that the notarization is invisible to the BC, although it might be advantageous to know that you are catching up and thus do not need to notarize.

Reply:
Yes, you are right. You would want to notarise the new block mined.